ocaml/xapi/xapi_xenops.ml

```diff
@@ -1651,8 +1651,11 @@ let update_vif ~__context id =
 							(match Pvs_proxy_control.find_proxy_for_vif ~__context ~vif with
 								| None -> ()
 								| Some proxy ->
-									debug "xenopsd event: Updating PVS_proxy for VIF %s.%s currently_attached <- %b" (fst id) (snd id) state.pvs_rules_active;
-									Db.PVS_proxy.set_currently_attached ~__context ~self:proxy ~value:state.pvs_rules_active
+									if Pvs_proxy_control.start_proxy ~__context vif proxy then begin
```

Comment:
This doesn't look right, because we want the proxy to be running already at this point (i.e. before the OVS rules are setup). And normally that would be the case. With this patch, we would configure the proxy twice.

We need to make sure that the rules are not installed when the proxy is not running, and then the state will come out correct at this point.